src/slack/slack_client.bal

```diff
@@ -0,0 +1,390 @@
+        return renameConversation(self.conversationClient, <@untainted> resolvedChannelId, newName);
+    }
+
+    # The `ConversationClient.listConversations()` function can be used to list all channels in a slack team.
+    #
+    # + return - An `error` if it is a failure or `Conversations` record if it is a success
+    public remote function listConversations() returns @tainted Conversations|error {
+        http:Client convClient = self.conversationClient;
+        var response = convClient->get(LIST_CONVERSATIONS_PATH);
+        if (response is error) {
+           return setResError(response);  
+        }
+        http:Response httpResp = <http:Response> response;
+        var jsonPayload = httpResp.getJsonPayload();
+        if (jsonPayload is error) {
+            return setJsonResError(jsonPayload);
+        }
+        json payload = <json> jsonPayload;
+        return mapConversationInfo(payload);
+    }
+
+    # The `ConversationClient.leaveConversation()` function can be used to leave a conversation.
+    #
+    # + channelName - Name of the conversation 
+    # + return - An 'error' if it is a failure or 'nil' if it is a success
+    public remote function leaveConversation(string channelName) returns @tainted error|() {
+        var resolvedChannelId = check self.resolveChannelId(self.idMap, channelName);
+        return leaveConversation(self.conversationClient, <@untainted> resolvedChannelId);
+    }
+
+    # The `ConversationClient.getConversationInfo()` function can be used to unarchive a conversation.
+    #
+    # + channelName - Name of the conversation to be created
+    # + includeLocale - Set this to `true` to receive the locale for this conversation. Defaults to `false`
+    # + memberCount - Set to `true` to include the member count for the specified conversation. Defaults to `false`
+    # + return - An `error` if it is a failure or the `Channel` record if it is a success
+    public remote function getConversationInfo(string channelName, boolean includeLocale = false, 
+                                    boolean memberCount = false) returns @tainted Channel|error {
+        var resolvedChannelId = check self.resolveChannelId(self.idMap, channelName);
+        return getConversationInfo(self.conversationClient, <@untainted> resolvedChannelId);
+    }
+
+    # The `ConversationClient.removeUserFromConversation()` function can be used to remove a user from a conversation.
+    #
+    # + channelName - Name of the conversation 
+    # + user - Name of the user to be removed
+    # + return - An `error` if it is a failure or `nil` if it is a success
+    public remote function removeUserFromConversation(string channelName, string user) returns @tainted error|() {
+        string userId = EMPTY_STRING;
+        string resolvedChannelId = check self.resolveChannelId(self.idMap, channelName);
+        userId = check getUserId(self.conversationClient, user); 
+        if (userId == EMPTY_STRING) {
```

Comment:
Why do we need to check whether the userId is `EMPTY_STRING` here? If `getUserId()` returns an error, that will be returned, and if it doesn't, we get the `userId` right? Please check my comment on `getUserId()` function too.